server/server.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express');
+const app = express();
+const port = 4000;
+const mongoose = require('mongoose');
+const db = require('../database/database.js');
+
+app.use('/:id', express.static(__dirname + '/../client/dist'));
+
+// open mongoose connection
+mongoose.connect('mongodb://localhost/fec_product_features', {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useFindAndModify: false,
+  useCreateIndex: true
+});
+
+mongoose.connection.on('error', console.error.bind(console, 'connection error:'));
```

Comment:
I am curious as to why you decided to move your mongoose.connect to the server file? Is it to ensure mongoose connects as soon as the server starts?

Reply:
Yes, exactly. I moved over the connection initiation to ensure it starts with the server. Admittedly, it was mainly done so to help with testing. Maybe this is something I can discuss with you and refactor it back to the database file without breaking tests.